cmd/root.go

```diff
@@ -106,13 +110,20 @@ func BuildFirewallConfiguration(options *RootOptions) (*iptables.FirewallConfigu
 		return nil, fmt.Errorf("--outgoing-proxy-port must be a valid TCP port number")
 	}
 
+	for _, subnet := range options.SubnetsToIgnore {
+		if !subnets.IsValid(subnet) {
+			return nil, fmt.Errorf("--subnets-to-ignore must be a comma separated list of subnets in CIDR notation")
```

Comment:
I'm in two minds about the error handling here. On the one hand, I think the way you've done it is concise and would work well, I personally think it looks great and clean. On the other though, I think we could be a bit more descriptive with the error.

If we have `[1.1.1.1/24, 0.0.0.0]` as an argument, we would simply be told the list has to be a comma separated list of subnets. In this case it would probably be easy to spot the mistake, but it probably wouldn't hurt to be explicit with which subnet was invalid ([here's a quick playground example I put together](https://play.golang.org/p/ZZdtPL-_T-A)).

My suggestion here would be to perhaps bring the `net` lib into scope in this file and just rely on `net.ParseCIDR` in this loop, it might simplify the code organisation a bit too, it feels a bit heavy to have a `subnets` package just for the validation function alone. wdyt @michaellzc?

Ultimately, there's no right or wrong here. I think what we have now is good, it's just to me it might be more likely for a subnet to have the wrong format. The argument itself will be documented anyway so presumably people will know it has to be a comma separated list. Hope it makes sense.

Reply:
@mateiidavid I was also debating whether or not to bring the "net" lib into `root.go`, but then I thought maybe I should just roll with the exiting convention like the `ports` module. Another problem I can think of with the `subnets module is error handling. I remember the go convention is returning a tuple `(value, err)`, but the `subnets.InValid` is not really following that convention. I am definitely not a go power user ;)

I think your way makes more sense, and I like the idea of being explicit about what's wrong with a specific CIDR. I'll make the change and look into the integration test. thanks for the feedback.